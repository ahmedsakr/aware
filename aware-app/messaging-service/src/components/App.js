import React, { Component } from 'react';
import '../style/App.css';
import '../../node_modules/bootstrap/dist/css/bootstrap.css'
import NavBar from './NavBar'
import ChatWindow from './ChatWindow'
import ChatTitle from './ChatTitle'
import DirectMessages from './DirectMessages'
import MessageInput from './MessageInput'

class App extends Component {
  render() {
    return (
      <div class="container-fluid" className="App">
        <div class="row">
          <div class="col-1">
            <NavBar />
          </div>

          <div class="col-8">
            <ChatTitle course="SYSC 2100" />
            <ChatWindow />
            <MessageInput />
          </div>

          <div class="col-3">
            <DirectMessages />
          </div>
        </div>
      </div>
    );
  }
}

export default App;