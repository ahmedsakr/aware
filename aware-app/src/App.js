import React, { Component } from 'react';

import Messenger from './messaging-service/Messenger'
import Landing from './landing/Landing'

class App extends Component {
    constructor() {
        super()
    
        this.state = {
          component: Landing
        }
    }

    render() {
        var Component = this.state.component;
        
        return (
            <Component loadMessenger = {this.loadMessenger}/>
        );
    }

    loadMessenger = () => {
        this.setState({
            component: Messenger
        })
    }
}

export default App;
