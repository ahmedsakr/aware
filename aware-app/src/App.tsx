import 'babel-polyfill'
import React, { Component } from 'react';
import io from 'socket.io-client'

import Messenger from './messaging-service/Messenger'
import Landing from './landing/Landing'

type AppState = {
    socket: SocketIOClient.Socket,
    component: string,
    username: string | null
};

export default class App extends Component<{}, AppState> {
    constructor(props: any) {
        super(props);

        this.state = {
          socket: io(),
          component: "landing",
          username: null
        };
    }

    shouldComponentUpdate(nextProps: any, nextState: any): boolean {
        return this.state.component !== nextState.component;
    }

    loadMessenger = (username: string): void => {
        this.setState({
            component: "messenger",
            username
        }); 
    }

    render(): JSX.Element {
        if (this.state.component === "landing") {
            return (
                <Landing
                    socket={this.state.socket}
                    loadMessenger = {this.loadMessenger} />
            );     
        } else {
            return (
                <Messenger
                    socket={this.state.socket}
                    username={this.state.username as string} />
            );
        }
    }
}
