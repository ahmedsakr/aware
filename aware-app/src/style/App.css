.aware-container {
  height: 100vh;
  max-height: 100vh;
  flex-direction: column;
}

.aware-column {
  height: 100%;
}

#messenger-root {
  flex-direction: column;
  height: 100%;
}

#navigation-header{
  flex: 0 1 auto;
}

#messenger-body {
  flex: 1 0 auto;
  display: flex;
}

#messenger {
  display: flex;
  flex-flow: column;
  justify-items: stretch;
}