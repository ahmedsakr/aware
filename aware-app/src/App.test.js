import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
import socket from 'socket.io-client'

jest.mock('socket.io-client');

describe('App', () => {
  it('should render correctly in "debug" mode', () => {

    const component = shallow(<App debug />);
    expect(component).toMatchSnapshot();
  });
});