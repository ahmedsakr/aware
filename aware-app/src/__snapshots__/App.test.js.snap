// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<Landing
  loadMessenger={[Function]}
  name={null}
  setUsername={[Function]}
  socket={
    Socket {
      "_callbacks": Object {
        "$connect": Array [
          [Function],
        ],
        "$connecting": Array [
          [Function],
        ],
      },
      "acks": Object {},
      "connected": false,
      "disconnected": true,
      "flags": Object {},
      "ids": 0,
      "io": Manager {
        "_callbacks": Object {
          "$close": Array [
            [Function],
          ],
          "$open": Array [
            [Function],
          ],
          "$packet": Array [
            [Function],
          ],
        },
        "_randomizationFactor": 0.5,
        "_reconnection": true,
        "_reconnectionAttempts": Infinity,
        "_reconnectionDelay": 1000,
        "_reconnectionDelayMax": 5000,
        "_timeout": 20000,
        "autoConnect": true,
        "backoff": Backoff {
          "attempts": 0,
          "factor": 2,
          "jitter": 0.5,
          "max": 5000,
          "ms": 1000,
        },
        "connecting": Array [
          [Circular],
        ],
        "decoder": Decoder {
          "reconstructor": null,
        },
        "encoder": Encoder {},
        "encoding": false,
        "engine": Socket {
          "_callbacks": Object {
            "$error": Array [
              [Function],
            ],
            "$open": Array [
              [Function],
            ],
          },
          "agent": false,
          "binaryType": null,
          "ca": null,
          "cert": null,
          "ciphers": null,
          "enablesXDR": false,
          "forceBase64": false,
          "forceJSONP": false,
          "forceNode": false,
          "hostname": "localhost",
          "id": null,
          "isReactNative": false,
          "jsonp": true,
          "key": null,
          "onlyBinaryUpgrades": undefined,
          "passphrase": null,
          "path": "/socket.io/",
          "perMessageDeflate": Object {
            "threshold": 1024,
          },
          "pfx": null,
          "pingInterval": null,
          "pingIntervalTimer": null,
          "pingTimeout": null,
          "pingTimeoutTimer": null,
          "policyPort": 843,
          "port": "80",
          "prevBufferLen": 0,
          "query": Object {},
          "readyState": "opening",
          "rejectUnauthorized": true,
          "rememberUpgrade": false,
          "secure": false,
          "timestampParam": "t",
          "timestampRequests": undefined,
          "transport": XHR {
            "_callbacks": Object {
              "$close": Array [
                [Function],
              ],
              "$drain": Array [
                [Function],
              ],
              "$error": Array [
                [Function],
              ],
              "$packet": Array [
                [Function],
              ],
            },
            "agent": false,
            "ca": null,
            "cert": null,
            "ciphers": null,
            "enablesXDR": false,
            "extraHeaders": undefined,
            "forceNode": false,
            "hostname": "localhost",
            "isReactNative": false,
            "key": null,
            "localAddress": undefined,
            "passphrase": null,
            "path": "/socket.io/",
            "pfx": null,
            "pollXhr": Request {
              "_callbacks": Object {
                "$data": Array [
                  [Function],
                ],
                "$error": Array [
                  [Function],
                ],
              },
              "agent": false,
              "async": true,
              "ca": null,
              "cert": null,
              "ciphers": null,
              "data": null,
              "enablesXDR": false,
              "extraHeaders": undefined,
              "index": 0,
              "isBinary": undefined,
              "key": null,
              "method": "GET",
              "passphrase": null,
              "pfx": null,
              "rejectUnauthorized": true,
              "requestTimeout": undefined,
              "supportsBinary": false,
              "uri": "http://localhost/socket.io/?EIO=3&transport=polling&t=MiVSrHm&b64=1",
              "xd": true,
              "xhr": XMLHttpRequest {
                "DONE": 4,
                "HEADERS_RECEIVED": 2,
                "LOADING": 3,
                "OPENED": 1,
                "UNSENT": 0,
                "abort": [Function],
                "addEventListener": [Function],
                "dispatchEvent": [Function],
                "getAllResponseHeaders": [Function],
                "getRequestHeader": [Function],
                "getResponseHeader": [Function],
                "handleError": [Function],
                "onreadystatechange": [Function],
                "open": [Function],
                "readyState": 1,
                "removeEventListener": [Function],
                "responseText": "",
                "responseXML": "",
                "send": [Function],
                "setDisableHeaderCheck": [Function],
                "setRequestHeader": [Function],
                "status": null,
                "statusText": null,
              },
              "xs": false,
            },
            "polling": true,
            "port": "80",
            "query": Object {
              "EIO": 3,
              "b64": 1,
              "t": "MiVSrHm",
              "transport": "polling",
            },
            "readyState": "opening",
            "rejectUnauthorized": true,
            "requestTimeout": undefined,
            "secure": false,
            "socket": [Circular],
            "supportsBinary": false,
            "timestampParam": "t",
            "timestampRequests": undefined,
            "xd": true,
            "xs": false,
          },
          "transportOptions": Object {},
          "transports": Array [
            "polling",
            "websocket",
          ],
          "upgrade": true,
          "upgrades": null,
          "writeBuffer": Array [],
        },
        "lastPing": null,
        "nsps": Object {
          "/": [Circular],
        },
        "opts": Object {
          "hostname": "localhost",
          "path": "/socket.io",
          "port": "80",
          "secure": false,
        },
        "packetBuffer": Array [],
        "readyState": "opening",
        "skipReconnect": false,
        "subs": Array [
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
          Object {
            "destroy": [Function],
          },
        ],
        "uri": "http://localhost",
      },
      "json": [Circular],
      "nsp": "/",
      "receiveBuffer": Array [],
      "sendBuffer": Array [],
      "subs": Array [
        Object {
          "destroy": [Function],
        },
        Object {
          "destroy": [Function],
        },
        Object {
          "destroy": [Function],
        },
      ],
    }
  }
/>
`;
